python-modules/chevah-python-test/test_python_binary_dist.py

```diff
@@ -3,99 +3,285 @@
+        for single_allowed_dep in allowed_deps:
+            if single_allowed_dep in single_actual_dep:
+                break
+        else:
+            unwanted_deps.append(single_actual_dep)
+    return unwanted_deps
+
+
+def main():
+    """
+    Launch tests to check required modules and OS-specific dependencies.
+    Exit with a relevant error code.
+    """
+    exit_code = 0
 
-# Linux specific modules.
-if platform_system == 'linux':
-    # On Linux we need spwd... but not on all Unix system, ex: AIX
     try:
-        import spwd
-        spwd
+        import zlib
+        zlib
     except:
-        print 'spwd missing.'
+        sys.stderr.write('"zlib" missing.\n')
         exit_code = 1
 
-sys.exit(exit_code)
+    try:
+        import _hashlib
+        _hashlib
+    except:
+        sys.stderr.write('standard "ssl" missing.\n')
+        exit_code = 2
+
+    try:
+        from OpenSSL import SSL, crypto, rand
+        SSL
+        crypto
+        rand
+    except:
+        sys.stderr.write('"OpenSSL" missing.\n')
+        exit_code = 3
+
+    try:
+        import Crypto
+        Crypto
+    except:
+        sys.stderr.write('"PyCrypto" missing.\n')
+        exit_code = 4
+
+    try:
+        import crypt
+        crypt
+    except:
+        sys.stderr.write('"crypt" missing.\n')
+        exit_code = 5
+
+    try:
+        import pysqlite2
+        pysqlite2
+    except:
+        sys.stderr.write('"pysqlite2" missing.\n')
+        exit_code = 6
+
+    try:
+        import setproctitle
+        setproctitle
+    except:
+        sys.stderr.write('"setproctitle" missing.\n')
+        exit_code = 7
+
+    try:
+        from ctypes import CDLL
+        CDLL
+    except:
+        sys.stderr.write('"ctypes - CDLL" missing.\n')
+        exit_code = 8
+
+    try:
+        from ctypes.util import find_library
+        find_library
+    except:
+        sys.stderr.write('"ctypes.utils - find_library" missing.\n')
+        exit_code = 9
+
+    try:
+        from Crypto.PublicKey import _fastmath
+        _fastmath
+    except:
+        sys.stderr.write('Crypto.PublicKey._fastmath missing. No GMP?\n')
+        exit_code = 10
+
+    # Windows specific modules.
+    if os.name == 'nt':
+        try:
+            from ctypes import windll
+            windll
+        except:
+            sys.stderr.write('"ctypes - windll" missing.\n')
+            exit_code = 11
+
+    if ( platform_system == 'linux' ) or ( platform_system == 'sunos' ):
+        # On Linux and Solaris we need spwd, but not on AIX or OS X.
+        try:
+            import spwd
+            spwd
+        except:
+            sys.stderr.write('spwd missing.\n')
+            exit_code = 12
+
+    # Compare the list of allowed deps for the current OS with the list of
+    # actual deps for the newly-built binaries returned by the script helper.
+    allowed_deps = get_allowed_deps()
+    if not allowed_deps:
+        sys.stderr.write('Got no allowed deps. Please check if {0} is a ' \
```

Comment:
don't use '\' for spliting long lines. you can splint string over long lines without any problems
```
sys.stderr.write(
    'Got no allowed deps. Please check if {0} is a '
    'supported operating system.\n'.format(platform.system()))